plutus-core/plutus-ir/src/PlutusIR/Transform/Inline.hs

```diff
@@ -144,29 +183,38 @@ processTerm
     :: Inlining tyname name uni fun a m
     => Term tyname name uni fun a
     -> m (Term tyname name uni fun a)
-processTerm = \case
-    v@(Var _ n) -> do
+processTerm = handleTerm <=< traverseOf termSubtypes applySubstitution where
+    handleTerm = \case
```

Comment:
I think it's helpful to give all local functions type signatures. Both for readability and because it helps pin down where the problem is when you change things.